tests/replace-manifest_test.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2015 The appc Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"encoding/json"
+	"os"
+	"testing"
+)
+
+func TestReplaceManifest(t *testing.T) {
+	workingDir := setUpTest(t)
+	defer cleanUpTest(workingDir)
+
+	manblob, err := detailedManifest.MarshalJSON()
```

Comment:
`schema.ImageManifest.MarshalJSON` calls `assertValid` for me though, to make sure that there isn't something horribly wrong with the manifest.